azurerm/internal/services/media/media_streaming_endpoint_resource.go

```diff
@@ -0,0 +1,591 @@
+	autoStart := utils.Bool(false)
+
+	parameters := media.StreamingEndpoint{
+		StreamingEndpointProperties: &media.StreamingEndpointProperties{
+			ScaleUnits: utils.Int32(int32(scaleUnits)),
+		},
+		Location: utils.String(location),
+	}
+
+	if _, ok := d.GetOk("auto_start_enabled"); ok {
+		autoStart = utils.Bool(d.Get("auto_start_enabled").(bool))
+	}
+	if _, ok := d.GetOk("access_control"); ok {
+		parameters.StreamingEndpointProperties.AccessControl = expandAccessControl(d)
+	}
+	if cdnEnabled, ok := d.GetOk("cdn_enabled"); ok {
+		parameters.StreamingEndpointProperties.CdnEnabled = utils.Bool(cdnEnabled.(bool))
+	}
+
+	if cdnProfile, ok := d.GetOk("cdn_profile"); ok {
+		parameters.StreamingEndpointProperties.CdnProfile = utils.String(cdnProfile.(string))
+	}
+
+	if cdnProvider, ok := d.GetOk("cdn_provider"); ok {
+		parameters.StreamingEndpointProperties.CdnProvider = utils.String(cdnProvider.(string))
+	}
+
+	if _, ok := d.GetOk("cross_site_access_policies"); ok {
+		parameters.StreamingEndpointProperties.CrossSiteAccessPolicies = expandCrossSiteAccessPolicies(d)
+	}
+
+	if _, ok := d.GetOk("custom_host_names"); ok {
+		customHostNames := d.Get("custom_host_names").([]interface{})
+		parameters.StreamingEndpointProperties.CustomHostNames = utils.ExpandStringSlice(customHostNames)
+	}
+
+	if description, ok := d.GetOk("description"); ok {
+		parameters.StreamingEndpointProperties.Description = utils.String(description.(string))
+	}
+
+	if maxCacheAge, ok := d.GetOk("max_cache_age_seconds"); ok {
+		parameters.StreamingEndpointProperties.MaxCacheAge = utils.Int64(int64(maxCacheAge.(int)))
+	}
+
+	future, err := client.Create(ctx, resourceGroup, accountName, streamingEndpointName, parameters, autoStart)
+	if err != nil {
+		return fmt.Errorf("Error creating Streaming Endpoint %q in Media Services Account %q (Resource Group %q): %+v", streamingEndpointName, accountName, resourceGroup, err)
+	}
+
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("Error waiting for creation of Streaming Endpoint %q in Media Services Account %q (Resource Group %q): %+v", streamingEndpointName, accountName, resourceGroup, err)
+	}
+
+	endpoint, err := client.Get(ctx, resourceGroup, accountName, streamingEndpointName)
+	if err != nil {
+		return fmt.Errorf("Error retrieving Streaming Endpoint %q from Media Services Account %q (Resource Group %q): %+v", streamingEndpointName, accountName, resourceGroup, err)
+	}
+
+	d.SetId(*endpoint.ID)
+
+	return resourceMediaStreamingEndpointRead(d, meta)
+}
+
+func resourceMediaStreamingEndpointUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Media.StreamingEndpointsClient
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	streamingEndpointName := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+	accountName := d.Get("media_services_account_name").(string)
```

Comment:
since this is the update, and these are immutable, we should instead parse these from the ID:

```suggestion
	id, err := parse.StreamingEndpoint(d.ID())
	if err != nil {
		return err
	}
```